src/server/bitops_family.cc

```
@@ -577,12 +593,488 @@ void BitCount(CmdArgList args, ConnectionContext* cntx) {
+  }
+
+ private:
+  ElementAccess access_;
+  EngineShard* shard_;
+};
+
+nonstd::expected<CommonAttributes, std::string> ParseCommonAttr(CmdArgParser& parser) {
+  CommonAttributes parsed;
+  using nonstd::make_unexpected;
+  if (!parser.HasAtLeast(2)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  auto encoding = parser.ToUpper().Next();
+  if (absl::StartsWith(encoding, "U")) {
+    parsed.type = EncodingType::UINT;
+  } else if (absl::StartsWith(encoding, "I")) {
+    parsed.type = EncodingType::INT;
+  } else {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  std::string_view bits = encoding;
+  bits = bits.substr(1);
+
+  if (!absl::SimpleAtoi(bits, &parsed.encoding_bit_size)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  if (parsed.encoding_bit_size <= 0 || parsed.encoding_bit_size > 64) {
+    return make_unexpected(
+        "invalid bitfield type. use something like i16 u8. note that u64 is not supported but i64 "
+        "is.");
+  }
+
+  if (parsed.encoding_bit_size == 64 && parsed.type == EncodingType::UINT) {
+    return make_unexpected(
+        "invalid bitfield type. use something like i16 u8. note that u64 is not supported but i64 "
+        "is.");
+  }
+
+  auto offset_proxy = parser.Next();
+  std::string_view offset_str = offset_proxy;
+  bool is_proxy = false;
+  if (absl::StartsWith(offset_str, "#")) {
+    offset_str = offset_str.substr(1);
+    is_proxy = true;
+  }
+  if (!absl::SimpleAtoi(offset_str, &parsed.offset)) {
+    return make_unexpected(kSyntaxErr);
+  }
+  if (is_proxy) {
+    parsed.offset = parsed.offset * parsed.encoding_bit_size;
+  }
+  return parsed;
+}
+
+nonstd::expected<int64_t, std::string> ParseValue(CmdArgParser& parser) {
+  using nonstd::make_unexpected;
+  if (!parser.HasAtLeast(1)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  auto value = parser.ToUpper().Next();
+  int64_t res{0};
+  if (!absl::SimpleAtoi(value, &res)) {
+    return make_unexpected(kSyntaxErr);
+  }
+  return res;
+}
```

This is just `parser.Next<int64_t>()`... Please see my previous comment about silent failures, because without it might be inconvenient to actually use the parser

---

Yes, this works here nicely, in fact, `ParseValue` can be eliminated all together(which I did).